### ISSUE TITLE
Presistent setup of Rocker with docker-compose

### ISSUE THREAD

**Issue by brennismagicgit** [rocker-org/rocker-versioned2#464]
Hello everyone, I am trying to setup a RStudio Server using rocker on my own server to host it via Caddy. I would like the data, packages, and so on to persist on my Server via bind mounts. However I get strange server startup behaviour when trying to mount local folders insider the `rocker/verse:4.1.3` conatiner.

My docker-compose currently looks like this.
```yaml
version: '3.6'
services:
  caddy:
    image: caddy:2-alpine
    restart: always
    volumes:
      - /data/caddy/data:/data
      - /data/caddy/config:/config
      - ./server/Caddyfile:/etc/caddy/Caddyfile
    networks:
      - rstudio_network
    ports:
      - 81:80
      - 444:443


  rstudio:
    image: rocker/verse:4.1.3
    volumes:
      - /data/rstudio/home:/home
      - /data/rstudio/local:/usr/local/lib/R
      - /data/rstudio/lib:/usr/lib/R
      - /data/rstudio/packages:/etc/rstudio
    restart: always
    depends_on:
      - caddy
    environment:
      - PASSWORD=$PASSWORD
      - ROOT=true
      - USERID=$USERID
      - USER=$USER
    networks:
      - rstudio_network
    ports:
      # Debug access 
      - 3000:8787

networks:
  rstudio_network:
    driver: bridge
```

the folder `data/rstudio` is fully owned by the user that I pass with env vars to my docker compose setup.
```bash
 USERID=1002 PASSWORD="xyz" USER="{PLACEHOLDER}" docker-compose up -d --force-recreate
```
as 
```shell
xxx@server:/data$ ls -lah
drwxrwxrwx  6 USER USER 4.0K May 26 17:20 rstudio
```
what file permissions do I need in order for the mount to succeed?
If I change the bind mount to a docker volume the server starts up good - although I do not prefer using volumes since they might be excidentely deleted ^^

Happy to hear you comments and happy to see how you host `rocker` via docker-compose.

**Comment by brennismagicgit** [comment]
Thanks for your answer! Caddy should not have any impact here.
What I am trying to achieve is basically that packages that are installed within the `rocker` container do not need to be reinstalled after each docker restart. I actually achieved this by mounting the local `/usr/local/lib/R/site-library` folder into the docker container with the `root` user / group.
I also figured out that overwriting the R installation via bind mount fails. You can only "override" the `site-library`

So the correct way would be.

```bash
  rstudio:
    image: rocker/verse:4.1.3
    volumes:
      - /data/rstudio/home:/home
      - /data/rstudio/site-library:/usr/local/lib/R/site-library
      - ./rstudio/rserver.conf:/etc/rstudio/rserver.conf
    restart: always
    depends_on:
      - caddy
    environment:
      - PASSWORD=$PASSWORD
      - ROOT=true
      - USERID=$USERID
      - USER=$USER
    networks:
      - rstudio_network
    ports:
      # Debug access 
      - 3000:8787
```
also overwriting the `/home` folder works.

Now that you mention it. Where can I find an example of the `./.rstudio_config:/home/rstudio/.config` file? Might be interesting for further configs. Other than that I think we can call this closed?

**Comment by eitsupi** [comment]
Containers created with docker compose are not destroyed until you explicitly run the `docker compose down` command, so you do not need to worry about reinstalling libraries.
When restarting Docker, the containers just stop (like `docker compose stop` command).

And, if you are rebuilding a container from a new image, the R and apt packages are expected to be updated on the new image, so you don't want to inherit the R packages from the old container, do you?

> Now that you mention it. Where can I find an example of the `./.rstudio_config:/home/rstudio/.config` file? Might be interesting for further configs. Other than that I think we can call this closed?

RStudio configuration files modified on the RStudio GUI are saved in `~/.config/rstudio`. (see #103)
In other words, nothing is saved here at first, but when we change the settings on RStudio, the configuration file is saved in this location.
It is useful to be able to make this file persistent, as it would be time-consuming to reconfigure the settings each time the container is rebuilt.
